### ISSUE TITLE
rustdoc v23 format (current nightly) is not supported

### ISSUE THREAD

**Issue by ehuss** [obi1kenobi/cargo-semver-checks#201]
### Steps to reproduce the bug with the above code

Ran `cargo semver-checks check-release` against https://github.com/rust-lang/mdBook/

### Actual Behaviour

After running for a while, I get the output:

```
    Updating index
     Parsing mdbook v0.4.22 (current)
     Parsing mdbook v0.4.21 (baseline)
Error: rustdoc format v23 for file /Users/eric/Proj/rust/mdBook/target/semver-checks/registry-mdbook-0_4_21/target/semver-checks/target/doc/mdbook.json is not supported
```

### Expected Behaviour

Doesn't error.

#### Software version

cargo-semver-checks 0.13.1

#### Operating system

macOS 12.6 (Darwin 21.6.0)

#### Command-line

```bash
/Users/eric/.cargo/bin/cargo-semver-checks semver-checks --bugreport
```

#### cargo version

```
> cargo -V
cargo 1.67.0-nightly (ba607b23d 2022-11-22)
```

#### Compile time information

- Profile: release
- Target triple: x86_64-apple-darwin
- Family: unix
- OS: macos
- Architecture: x86_64
- Pointer width: 64
- Endian: little
- CPU features: cmpxchg16b,fxsr,llvm14-builtins-abi,sse,sse2,sse3,ssse3
- Host: x86_64-apple-darwin

**Comment by obi1kenobi** [comment]
Thanks for the report. I believe rustdoc format v23 just recently got released in nightly, and isn't supported at the moment.

In general, rustdoc formats can be released much more frequently than we can keep up with, so given limited time and resources, only most-recent stable and beta Rust are currently guaranteed to be supported. Nightly rustdoc formats are "best effort."

Most often, adding support for the new format is pretty seamless and I can do it quickly and easily. Unfortunately, I believe v23 is the version that stopped differentiating between "methods" and "functions" in the rustdoc, so we might need to do a bit of work on our lints and test suite to make everything work with it.

I think that v23 or something directionally similar will probably be in the next beta, so we'll get started on that work. I unfortunately don't have a timeline yet.

**Comment by obi1kenobi** [comment]
The fix for this will be in the next release, sometime in the next week or so (on or before the next Rust version release).

If this issue is time-sensitive on your end, I could push a new `cargo-semver-checks` release in the morning.

Thanks for bearing with us!

**Comment by obi1kenobi** [comment]
v0.14.0 is out now and supports rustdoc v23. Thanks again for bearing with me.
